src/analysis/normed_space/star/multiplier.lean

```diff
@@ -0,0 +1,479 @@
+def coe_hom [star_ring 𝕜] [star_ring A] [star_module 𝕜 A] [normed_star_group A] :
+  A →⋆ₙₐ[𝕜] 𝓜(𝕜, A) :=
+{ to_fun := λ a, a,
+  map_smul' := λ k a, by {ext; simp only [coe_fst, coe_snd, continuous_linear_map.map_smul,
+    smul_fst, smul_snd]},
+  map_zero' := by {ext; simp only [coe_fst, coe_snd, map_zero, zero_fst, zero_snd]},
+  map_add' := λ a b, by {ext; simp only [coe_fst, coe_snd, map_add, add_fst, add_snd]},
+  map_mul' := λ a b, by {ext; simp only [coe_fst, coe_snd, mul_apply', flip_apply, mul_fst, mul_snd,
+    continuous_linear_map.coe_mul, function.comp_app, mul_assoc]},
+  map_star' := λ a, by {ext; simp only [coe_fst, coe_snd, mul_apply', star_fst, star_snd,
+    flip_apply, star_mul, star_star]} }
+
+/-!
+### Norm structures
+We define the norm structure on `𝓜(𝕜, A)` as the pullback under
+`double_centralizer.add_group_hom_to_prod : 𝓜(𝕜, A) →+ (A →L[𝕜] A) × (A →L[𝕜] A)`, which provides
+a definitional isometric embedding. Consequently, completeness of `𝓜(𝕜, A)` is obtained by proving
+that the range of this map is closed.
+
+In addition, we prove that `𝓜(𝕜, A)` is a normed algebra, and, when `A` is a C⋆-algebra, we show
+that `𝓜(𝕜, A)` is also a C⋆-algebra. Moreover, in this case, for `a : 𝓜(𝕜, A)`,
+`‖a‖ = ‖a.fst‖ = ‖a.snd‖`. -/
+
+/-- The normed group structure is inherited as the pullback under the additive group monomoprhism
+`double_centralizer.to_prod : 𝓜(𝕜, A) →+ (A →L[𝕜] A) × (A →L[𝕜] A)` -/
+instance : normed_add_comm_group 𝓜(𝕜, A) :=
+normed_add_comm_group.induced _ _ (to_prod_hom : 𝓜(𝕜, A) →+ (A →L[𝕜] A) × (A →L[𝕜] A)) ext
+
+lemma norm_def (a : 𝓜(𝕜, A)) : ‖a‖ = ‖a.to_prod‖ := rfl
+lemma nnnorm_def (a : 𝓜(𝕜, A)) : ‖a‖₊ = ‖a.to_prod‖₊ := rfl
+
+instance : normed_space 𝕜 𝓜(𝕜, A) :=
+{ norm_smul_le := λ k a, (norm_smul k a.to_prod).le,
+  .. double_centralizer.module }
+
+lemma uniform_embedding_to_prod : uniform_embedding (@to_prod 𝕜 A _ _ _ _ _) :=
+uniform_embedding_comap ext
+
+instance [complete_space A] : complete_space 𝓜(𝕜, A) :=
+begin
+  rw complete_space_iff_is_complete_range uniform_embedding_to_prod.to_uniform_inducing,
+  apply is_closed.is_complete,
+  simp only [range_to_prod, set.set_of_forall],
+  refine is_closed_Inter (λ x, is_closed_Inter $ λ y, is_closed_eq _ _),
+  exacts [((continuous_linear_map.apply 𝕜 A _).continuous.comp continuous_snd).mul continuous_const,
+    continuous_const.mul ((continuous_linear_map.apply 𝕜 A _).continuous.comp continuous_fst)],
+end
+
+noncomputable instance : normed_ring 𝓜(𝕜, A) :=
+{ norm_mul := λ a b,
+    begin
+      refine max_le ((norm_mul_le _ _).trans _) ((norm_mul_le _ _).trans _),
+      exact mul_le_mul (le_max_left _ _) (le_max_left _ _) (norm_nonneg _)
+        ((norm_nonneg _).trans $ le_max_left _ _),
+      exact mul_comm (‖a.snd‖) (‖b.snd‖) ▸ mul_le_mul (le_max_right _ _) (le_max_right _ _)
+        (norm_nonneg _) ((norm_nonneg _).trans $ le_max_right _ _),
+    end,
+  .. double_centralizer.ring,
+  .. double_centralizer.normed_add_comm_group }
```

Comment:
Doesn't this do the right thing?
```suggestion
noncomputable instance : normed_ring 𝓜(𝕜, A) :=
normed_ring.induced _ _ (to_prod_mul_opposite_hom : 𝓜(𝕜, A) →+* (A →L[𝕜] A) × (A →L[𝕜] A)) ext
```
Note, you will have to define `to_prod_mul_opposite_hom`.

Reply:
Two issues: 
1. This doesn't work because there is no norm on `ᵐᵒᵖ` to pull back.
2. Even if it did, I think we would be putting two different norm structures (because `ᵐᵒᵖ` is irreducible) on `𝓜(𝕜, A)`, because we get `normed_add_comm_group` from `to_prod`.
3. We could put a norm on `ᵐᵒᵖ` and then *only* use `normed_ring.induced` but then we get annoying defeqs for the norm as `max ∥a.fst∥ ∥mul_opposite.op a.snd∥`.

Reply:
I'll PR 1.

Reply:
okay, after #18470 and #18471 are in I'll merge master and use this instead.

Reply:
Do you need https://github.com/leanprover-community/mathlib/pull/18471 too?

Reply:
Because I also need completeness, which I get by showing the range of `to_prod` is closed, but now I'll need to switch to `to_prod_mul_opposite`. I mean, I guess I could still go through `to_prod` if needed.

Reply:
Well, I guess it's lucky I added it then!

Reply:
@eric-wieser and @YaelDillies this is done now. I left in some facts about `to_prod` even though they are no longer used, but if you want me to get rid of them I can (e.g, `range_to_prod` and `to_prod_hom`).